### DISCUSSION TITLE
[explore] Tweaks to sorting in metric search

### DISCUSSION THREAD

**Discussion by zuzana-vej** [apache/superset#18470] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently when users searches metric in Chart Explore Metric dropdown, the search first displays the "ending" string of the metric, even if there is an exact match. For example if you search for `san_francisco` and your metrics include `population_in_san_francisco` `san_francisco` and `san_francisco_weather` your first result would be `population_in_san_francisco`

The proposal is to at least put the exact match first. See below more on ideal experience.

**Describe the solution you'd like**

Ideal experience?
1. exact match
2. starting string matches
3. anything else (string in middle or end)

From above example the expected result would be
`san_francisco`
`san_francisco_weather`
`population_in_san_francisco`

**Comment by junlincc** [comment]
@zuzana-vej thanks for suggesting! it makes perfect sense.  It is a good-first-issue, but in case no one picks it up, we should be able to squeeze it in when enhancing the metric component.
